### PR TITLE
Fixes various Super Soaker issues

### LINKED ISSUE #22134: Shooting a fuel tank with a super soaker causes it to explode
It's a light stream of water hitting a fuel tank, not a bullet.

#### Steps to reproduce
Get super soaker (/obj/item/weapon/gun/siren/supersoaker) from the beach_party random vault (pumping not required).
Shoot fuel tank.
Explode.

### LINKED ISSUE #23661: super soakers blow up weldertanks
(WEB REPORT BY: jeremopolis REMOTE: 206.221.180.138:7777)
# Revision
b27e536b35093cab5f736df005aa16a508b4d2cc

# Description
super soakers blow up weldertanks, this should not happen.
# Steps to Reproduce
shoot a weldertank with a supersoaker filled with water.
# What you Expected
that it would not blow up(it was hit by accident)
# What Actually Happened
it blew up and killed someone. how about nah

### PR DESCRIPTION
**Pull request by DeityLink**
Fixes #22134
Fixes #23661

image

:cl:
* bugfix: Super Soakers now actually wet what they hit with the reagents they contain, as well as the floor bellow them.
* bugfix: Super Soakers no longer blow up welding fuel tanks, nor do any projectile that deals 0 damage.
* imageadd: Added in-hand icons to the Super Soaker and Squirt Pistol.

### CODE REVIEW

File: code/modules/reagents/reagent_dispenser.dm [hunk 1]
```diff
@@ -168,7 +168,7 @@
 
 /obj/structure/reagent_dispensers/fueltank/bullet_act(var/obj/item/projectile/Proj)
 	if(istype(Proj ,/obj/item/projectile/beam)||istype(Proj,/obj/item/projectile/bullet)||istype(Proj,/obj/item/projectile/ricochet))
-		if(!istype(Proj ,/obj/item/projectile/beam/lasertag) && !istype(Proj ,/obj/item/projectile/beam/practice) )
+		if(!istype(Proj ,/obj/item/projectile/beam/lasertag) && !istype(Proj ,/obj/item/projectile/beam/practice) && Proj.get_damage())
```
Comment (Kurfursten) [inline code/modules/reagents/reagent_dispenser.dm]: shouldn't the rest of this ifcheck be redundant now?

Comment (DeityLink) [reply code/modules/reagents/reagent_dispenser.dm]: nope, the lasertag and practice laser guns deal "holo" damage which still counts as damage.

Comment (Kurfursten) [reply code/modules/reagents/reagent_dispenser.dm]: maybe check to see if damage type is holo then

Comment (DeityLink) [reply code/modules/reagents/reagent_dispenser.dm]: nvm looks like you were right, no idea where I remember practice laser gun holo damage from then. I could swear someone lasered me with it years ago which made me fall into "fake crit" for a few seconds.

**Pull request merged**